core/kernel/persistence/smoothsql/search/QueryJoiner.php

```diff
@@ -0,0 +1,304 @@
+                 $this->getDriverEscaper()->reserved('ORDERJ') . '.' . $this->getDriverEscaper()->reserved('subject') . ' , ' .
+                 $this->getDriverEscaper()->reserved('ORDERJ') . '.' . $this->getDriverEscaper()->reserved('object') . ' ' .
+                 $this->getDriverEscaper()->dbCommand('FROM') . ' ( ' .
+                 $join . ')' . 
+                 $this->getDriverEscaper()->dbCommand('AS') . ' ' . $this->getDriverEscaper()->reserved('SUBJ') . ' ' .
+                 $this->getDriverEscaper()->dbCommand('JOIN') . ' ( ' . 
+                 $this->getDriverEscaper()->dbCommand('SELECT') . ' ' . $this->getDriverEscaper()->reserved('subject') . ' , ' .
+                 $this->getDriverEscaper()->reserved('object')  . ' ' . $this->getDriverEscaper()->dbCommand('FROM') . ' ' .
+                 $this->getDriverEscaper()->reserved('statements') . ' ' . $this->getDriverEscaper()->dbCommand('WHERE') . ' ' .
+                 $this->getDriverEscaper()->reserved('predicate') . ' = ' . 
+                 $this->getDriverEscaper()->quote($this->getDriverEscaper()->escape($sortParam)) . ' ' .
+                 $this->getLanguage() . ' ' . $this->getDriverEscaper()->dbCommand('GROUP') . ' ' . $this->getDriverEscaper()->dbCommand('BY') . ' ' .
+                 $this->getDriverEscaper()->reserved('subject') . ' , ' .
+                 $this->getDriverEscaper()->reserved('object')  . ' ) ' . 
+                 $this->getDriverEscaper()->dbCommand('AS') . ' ' . $this->getDriverEscaper()->reserved('ORDERJ') . ' ' . 
+                 $this->getDriverEscaper()->dbCommand('ON') . ' ( ' . 
+                 $this->getDriverEscaper()->reserved('SUBJ') . '.' . $this->getDriverEscaper()->reserved('subject') . ' = ' .
+                 $this->getDriverEscaper()->reserved('ORDERJ') . '.' . $this->getDriverEscaper()->reserved('subject') . ' ) ) ' .
+                 $this->getDriverEscaper()->dbCommand('AS') . ' ' . $this->getDriverEscaper()->reserved('J') . ' ' .
+                 $this->getDriverEscaper()->dbCommand('ON') . ' ( ' . 
+                 $this->getDriverEscaper()->reserved('J') . '.' . $this->getDriverEscaper()->reserved('subject') . ' = ' .
+                 $this->getDriverEscaper()->reserved('R') . '.' . $this->getDriverEscaper()->reserved('object') . ' ) ' .
+                 $this->getDriverEscaper()->dbCommand('GROUP') . ' ' . $this->getDriverEscaper()->dbCommand('BY') . ' ' .
+                 $this->getDriverEscaper()->reserved('T') . '.' . $this->getDriverEscaper()->reserved('subject') . ' ' .
+                 $this->getDriverEscaper()->dbCommand('ORDER') . ' ' . $this->getDriverEscaper()->dbCommand('BY') . ' ' .
+                 $this->getDriverEscaper()->reserved('sorter') . ' ' . $this->getDriverEscaper()->dbCommand($sortSens) . ' ) ' .
+                 $this->getDriverEscaper()->dbCommand('AS') . ' ' . $this->getDriverEscaper()->reserved('rootq');
```

Comment:
Finally, depending on the existing implementation Join or Left Join for the sorter ?